### ISSUE TITLE
Socket npm Unable to prompt to accept risk, need TTY to do so

### ISSUE THREAD

**Issue by Dotnaught** [SocketDev/socket-cli-js#32]
MacOS, 13.2.1
Node 16.17.1

Installing Socket CLI, aliasing socket npm to npm, and attempting to create a sample demo app using the similarly named but unrelated Socket Runtime (https://sockets.sh/) produces the error below.

The error also occurs without `alias npm="socket npm"` if you enter `socket npm create socket-app svelte`.

The app template can be success created if Socket CLI is installed but not aliased and not invoked:  `npm create socket-app svelte`

```
npm create socket-app svelte
Initializing npm package...Ok.
Installing developer dependencies...
Unable to run npm install: Error: Command failed: npm install -D vite @sveltejs/vite-plugin-svelte
esbuild@0.17.12 (​https://socket.dev/npm/package/esbuild/overview/0.17.12​) contains risks:
  Install scripts - Install scripts are run when the package is installed. The majority of malware in npm is hidden in install scripts.
npm ERR! Socket npm Unable to prompt to accept risk, need TTY to do so

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/xxx/.npm/_logs/2023-03-18T17_19_30_566Z-debug-0.log

    at ChildProcess.exithandler (node:child_process:400:12)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1093:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)
npm ERR! code 1
npm ERR! path /Users/tom/Code/ghub/socketrss
npm ERR! command failed
npm ERR! command sh -c create-socket-app svelte

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/xxx/.npm/_logs/2023-03-18T17_19_29_740Z-debug-0.log
```

And the verbose info:
```
 verbose stack Error: command failed
38 verbose stack     at ChildProcess.<anonymous> (/Users/xxx/.nvm/versions/node/v16.17.1/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:53:27)
38 verbose stack     at ChildProcess.emit (node:events:513:28)
38 verbose stack     at maybeClose (node:internal/child_process:1093:16)
38 verbose stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)
39 verbose cwd /Users/xxx/xxx/xxx/socketapp
40 verbose Darwin 22.3.0
41 verbose node v16.17.1
42 verbose npm  v9.2.0
43 error code 1
44 error path /Users/xxx/xxx/xxx/socketapp
45 error command failed
46 error command sh -c create-socket-app svelte
47 verbose exit 1
```

**Comment by bmeck** [comment]
This is because the `npm install -D vite @sveltejs/vite-plugin-svelte` spawned by `socket-app` doesn't provide an interactive terminal using `stdio: 'inherit'`. Since there is no way for a person to respond to the prompt, the CLI rejects anything that would be significant enough to cause a prompt. We do not have an expected UX to allow for installing without being able to respond to prompts at this time. We do still print out the issue and hope to provide an accept list functionality per package soon. My personal recommendation would be for [`create-socket-app`](https://github.com/socketsupply/create-socket-app/blob/main/index.js) to be modified so that STDIN isn't disconnected during that subprocess for now.

**Comment by Dotnaught** [comment]
Thanks, I've cross-posted your recommendation.

**Comment by bmeck** [comment]
@Dotnaught this might be fixed now w/ the 0.6.0 release w/o any action on their part
